packages/react-dom/src/events/__tests__/EnterLeaveEventPlugin-test.js

```diff
@@ -9,56 +9,53 @@
 
 'use strict';
 
-var EnterLeaveEventPlugin;
 var React;
 var ReactDOM;
-var ReactDOMComponentTree;
-var ReactTestUtils;
+
+const simulateMouseEvent = (node, type, relatedTarget) => {
+  node.dispatchEvent(
+    new MouseEvent(type, {
+      bubbles: true,
+      cancelable: true,
+      relatedTarget,
+    }),
+  );
+};
 
 describe('EnterLeaveEventPlugin', () => {
   beforeEach(() => {
     jest.resetModules();
 
     React = require('react');
     ReactDOM = require('react-dom');
-    ReactTestUtils = require('react-dom/test-utils');
-    // TODO: can we express this test with only public API?
-    ReactDOMComponentTree = require('../../client/ReactDOMComponentTree');
-    EnterLeaveEventPlugin = require('../EnterLeaveEventPlugin');
   });
 
   it('should set relatedTarget properly in iframe', () => {
-    var iframe = document.createElement('iframe');
+    const iframe = document.createElement('iframe');
     document.body.appendChild(iframe);
-
-    var iframeDocument = iframe.contentDocument;
+    const iframeDocument = iframe.contentDocument;
 
     iframeDocument.write(
       '<!DOCTYPE html><html><head></head><body><div></div></body></html>',
     );
     iframeDocument.close();
 
-    var component = ReactDOM.render(
-      <div />,
+    let events = [];
+    const component = ReactDOM.render(
+      <div
+        onMouseOver={e => {
+          e.persist();
+          events.push(e);
+        }}
+      />,
       iframeDocument.body.getElementsByTagName('div')[0],
     );
-    var div = ReactDOM.findDOMNode(component);
 
-    var extracted = EnterLeaveEventPlugin.extractEvents(
-      'topMouseOver',
-      ReactDOMComponentTree.getInstanceFromNode(div),
-      {target: div},
-      div,
-    );
-    expect(extracted.length).toBe(2);
+    simulateMouseEvent(component, 'mouseover', iframe.contentWindow);
 
-    var leave = extracted[0];
-    var enter = extracted[1];
-
-    expect(leave.target).toBe(iframe.contentWindow);
-    expect(leave.relatedTarget).toBe(div);
-    expect(enter.target).toBe(div);
-    expect(enter.relatedTarget).toBe(iframe.contentWindow);
+    expect(events.length).toBe(1);
+    expect(events[0].target).toBe(component);
+    expect(events[0].relatedTarget).toBe(iframe.contentWindow);
```

Comment:
Which part of the code are this testing here? Are we confident that removing it causes test to fail?